src/overflow/depression_fill.py

```diff
@@ -0,0 +1,255 @@
+from heapq import heappush, heappop
+from enum import Enum
+import numpy as np
+from numba import njit
+from osgeo import gdal
+from overflow.fill_pits import (
+    priority_flood_tile,
+    handle_edge,
+    handle_corner,
+    make_sides,
+)
+from overflow.setup_bands import setup_bands
+from overflow.util.raster import RasterChunk, raster_chunker, read_tile
+
+
+class Coordinate:
+    row: np.int32
+    col: np.int32
+
+    def __init__(self, row, col):
+        self.row = row
+        self.col = col
+
+
+class Side(Enum):
+    TOP = 1
+    RIGHT = 2
+    BOTTOM = 3
+    LEFT = 4
+
+
+class Corner(Enum):
+    TOP_LEFT = 1
+    TOP_RIGHT = 2
+    BOTTOM_RIGHT = 3
+    BOTTOM_LEFT = 4
+
+
+class DemLabelArr:
+    labels: np.ndarray
+    dem: np.ndarray
+
+
+class DemLabel:
+    label: int
+    dem: float
+
+
+@njit
+def depression_fill(input_path, output_path, chunk_size):
+    """
+    Fill depressions in a digital elevation model (DEM).
+    Algorithm 3 from Parallel Priority-Flood (R. Barnes, 2016)
+    """
+    with setup_bands(input_path, output_path) as bands:
+        input_band, output_band, no_data_value = bands
+        n_chunks_row, n_chunks_col = calculate_chunks(input_band, chunk_size)
+
+        for tile_index, tile_dem in enumerate(raster_chunker(input_band, chunk_size)):
+            sides = create_tile_side_flags(tile_dem, n_chunks_row, n_chunks_col)
+            tile_dem_labels, _ = priority_flood_tile(tile_dem, sides, no_data_value)
```

Comment:
don't we need to save the flooded dem tile here and use that as input to raise elevations? I think the logic is that there could be depressions fully interior to the tile that would be missed if we simply raise all cells to the minimum from the elevation graph